hikari/channels.py

```diff
@@ -1696,6 +1696,90 @@ class GuildThreadChannel(TextableGuildChannel):
     Will be `None` for threads created before 2020-01-09.
     """
 
+    async def edit(
+        self,
+        *,
+        name: undefined.UndefinedOr[str] = undefined.UNDEFINED,
+        archived: undefined.UndefinedOr[bool] = undefined.UNDEFINED,
+        auto_archive_duration: undefined.UndefinedOr[time.Intervalish] = undefined.UNDEFINED,
+        locked: undefined.UndefinedOr[bool] = undefined.UNDEFINED,
+        invitable: undefined.UndefinedOr[bool] = undefined.UNDEFINED,
+        rate_limit_per_user: undefined.UndefinedOr[time.Intervalish] = undefined.UNDEFINED,
+        flags: undefined.UndefinedOr[ChannelFlag] = undefined.UNDEFINED,
+        applied_tags: undefined.UndefinedOr[typing.Sequence[ForumTag]] = undefined.UNDEFINED,
+        reason: undefined.UndefinedOr[str] = undefined.UNDEFINED,
+    ) -> PartialChannel:
```

Comment:
We cant abide by the issue you linked this PR to yet, as it will (as you found out) create conflicts on mypy. The best to do here is to just temporarily add the thread fields to `GuildChannel.edit`. You can also re-scope this PR and complete that issue if you wish, by properly splitting the arguments in the different objects, which would fix the mypy issue

Reply:
That's why I left things as they are in the PR. I wasn't sure if someone might have a better idea of how to approach it.

One thought I had is that threads, dms, and channels are all distinct types and only share some behavior. So one idea I had was maybe to create a base channel type for the shared behavior, then branch that to dm channel, guild channel, and thread subtypes that implement the respective behaviors. That's a more extensive change though which is beyond the scope of what I should be doing without input. Will almost certainly be a breaking change for someone.

So, I'm gonna merge the thread behavior into the `GuildChannel.edit` and remove the `GuildThreadChannel.edit`. I would be open to helping break up the channel types, though, if there were interest in going through that process.

Reply:
>  I would be open to helping break up the channel types, though, if there were interest in going through that process.

Yeah, that would be nice :)